app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,27 +5,14 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
-  end
-
-  def create
-    site_banner = current_publisher.site_banner || SiteBanner.new
-    donation_amounts = JSON.parse(sanitize(params[:donation_amounts]))
-    site_banner.update(
-      publisher_id: current_publisher.id,
-      title: sanitize(params[:title]),
-      donation_amounts: donation_amounts,
-      default_donation: donation_amounts.second,
-      social_links: params[:social_links].present? ? JSON.parse(sanitize(params[:social_links])) : {},
-      description: sanitize(params[:description])
-    )
-    head :ok
-  end
-
-  def fetch
-    if current_publisher.site_banner.present?
-      data = current_publisher.site_banner.read_only_react_property
+  def show
+    if params[:default]
```

Comment:
Can we extract this out to a method called like `def site_banner`. The implementation would look something like this with memoization.

```ruby
def site_banner
  @site_banner ||= 
    if params[:default]
      current_publisher.default_site_banner
    else 
      current_publisher.site_banners.find_by(id: params[:id])
    end
end
```